water-gui.py

```diff
@@ -197,6 +197,18 @@ def analyze_menubutton(self, var: IntVar, menu: Menubutton) -> Optional[str]:
 
 (Error: '{str(e)}')""")
             return None
+    
+    def no_entry(self, p: str, v: str, u: float) -> str:
+        "register a sale, but not a monetary entry."
+        ask_for_reason = ReasonEntry()

```

Comment:
This will be added inside of `tools.prefabricated`. It will be a separate Tk app to prompt for a reason.